### ISSUE TITLE
Add standalone pending state provider function

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#2058]
### Describe the feature

getting the `Pending` state as `StateProvider` currently requires:

1. find block hash of target block in tree
2. get `PostStateDataProvider` via 1.)
3. get `StateProvider` via 2.)

For RPC the ideal DX would be:

```rust
BlockId::Number(BlockNumberOrTag::Pending) => self.<>.pending()
```
ref
https://github.com/paradigmxyz/reth/blob/b9bab335e6834e0760bbab8fc5163b7e405afd39/crates/rpc/rpc/src/eth/api/transactions.rs#L93-L107

If you do this independently of a specific block, you could use self-created local blocks instead of blocks from CL.

**Comment by rakita** [comment]
Created getter for pending blocks, those are blocks that directly extend the canonical chain: https://github.com/paradigmxyz/reth/pull/2091

**Comment by mattsse** [comment]
closed via #2294
